File: backend/src/genres/genres.controller.ts
import { Controller, Get } from '@nestjs/common';
import { GenresService } from './genres.service';

@Controller('genres')
export class GenresController {
  constructor(private readonly genresService: GenresService) {}

  @Get()
  async fetchGenres() {
    return this.genresService.fetchGenres();
  }
}

File: backend-java/src/main/java/com/laura/backendjava/controller/GenresController.java
package com.laura.backendjava.controller;

import com.laura.backendjava.service.GenresService;
import com.laura.backendjava.client.tmdb.dto.TmdbGenresResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/genres")
public class GenresController {
    private final GenresService genresService;

    public GenresController(GenresService genresService) {
        this.genresService = genresService;
    }

    @GetMapping
    public TmdbGenresResponse fetchGenres() {
        return genresService.fetchGenres();
    }
}